File: script/bookworm.py
# Install required modules
import requests
import pyscript

# Define functions
def fetch_data_from_php(url):
    response = requests.get(url)
    if response.status_code == 200:
        return response.json()
    else:
        print('Error fetching data from example.php.')
        return None

def update_question_and_answers(data, cnt):
    if len(data) > cnt:
        question_label.text = data[cnt]['question']
        answer[0] = data[cnt]['answer1']
        answer[1] = data[cnt]['answer2']
        shuffle_array(answer)
        btn1.text = answer[0]
        btn2.text = answer[1]
        global correct_answer
        correct_answer = data[cnt]['answer']
    else:
        print('No more questions available.')

def shuffle_array(array):
    for i in range(len(array) - 1, 0, -1):
        j = random.randint(0, i)
        array[i], array[j] = array[j], array[i]

def compare_if_correct(answer):
    if answer == correct_answer:
        print(True)
    else:
        print(False)

# Specify the URL for 'example.php'
example_php_url = '../html/example.php'  # Replace with the actual URL

# Fetch data from 'example.php'
data = fetch_data_from_php(example_php_url)

if data:
    # Initialize variables
    answer = [None, None]
    correct_answer = None
    cnt = 0

    # Execute functions
    update_question_and_answers(data, cnt)

    # Define button click events
    def btn1_click():
        compare_if_correct(btn1.text)
        global cnt
        cnt += 1
        update_question_and_answers(data, cnt)

    def btn2_click():
        compare_if_correct(btn2.text)
        global cnt
        cnt += 1
        update_question_and_answers(data, cnt)

    # Simulate button clicks
    btn1_click()
    btn2_click()

File: script/bookworm.js
$(document).ready(function () {
    var buttons = document.getElementsByClassName('choice-button');
    var questionLabel = $('#question-label');
    var score = 0;
    var lives = 3;
    var correctAnswer;
    let answer = new Array();
    var message;
    var game = 1;
    function updateQuestionAndAnswers(data) {
        if (data.length > cnt && cnt<10) {
            $("#question-label").text(data[cnt].question);
            answer[0] = data[cnt].answer1;
            answer[1] = data[cnt].answer2;
            shuffleArray(answer);
            $("#btn1").text(answer[0]);
            $("#btn2").text(answer[1]);
            correctAnswer = data[cnt].answer
        } else {
            showScore(score,lives);
        }
    }
    
    function shuffleArray(array) {
        for (let i = array.length - 1; i > 0; i--) {
            const j = Math.floor(Math.random() * (i + 1));
            [array[i], array[j]] = [array[j], array[i]];
        }
    }
    
    function CompareIfCorrect(answer){
        if (answer == correctAnswer){
            reduceLives(".life-right");
            score++;
        }else{
            reduceLives(".life");
            lives--;
        }
    }
    function reduceLives(life) {
        var lives = $(life);
        var remainingLives = lives.filter(':visible');
        
        if (remainingLives.length > 1) {
            remainingLives.last().hide();
        } else {
            showScore(score,lives);
        }
    }

    function showScore(score,lives) {
        if(lives>0)
        {
            message = "Congratulations!";
        }
        else
        {
            message = "You Died!";
        }
        location.replace("congrats.php?score=" + score + "&message=" + message+"&game="+game);
    }

    
    var cnt = 0;
    $.ajax({
            url: '../html/getQuestions.php',
            type: 'GET',
            dataType: 'json',
            success: function (data) {
            
                updateQuestionAndAnswers(data);
                
                $('#btn1').click(function () {
                    CompareIfCorrect($("#btn1").text());
                    cnt++;
                    updateQuestionAndAnswers(data); 
                });
                
                $('#btn2').click(function () {
                    CompareIfCorrect($("#btn2").text());
                    cnt++;
                    updateQuestionAndAnswers(data); 
                });
            },
            error: function (data) {
                console.error('Error fetching new questions and answers. ');
            }
        });
});